test/Generics/associated_type_typo.swift

```diff
@@ -14,33 +14,31 @@ protocol P2 {
 protocol P3 { }
 protocol P4 { }
 
-// expected-error@+1{{'assoc' is not a member type of 'T'}}
+// expected-error@+1{{'T' does not have a member type named 'assoc'; did you mean 'Assoc'?}}
 func typoAssoc1<T : P1>(x: T.assoc, _: T) { } 
 
 // expected-error@+2{{'T' does not have a member type named 'assoc'; did you mean 'Assoc'?}}{{53-58=Assoc}}
 // expected-error@+1{{'U' does not have a member type named 'assoc'; did you mean 'Assoc'?}}{{64-69=Assoc}}
 func typoAssoc2<T : P1, U : P1>(_: T, _: U) where T.assoc == U.assoc {}
 
 // CHECK-GENERIC-LABEL: .typoAssoc2
-// CHECK-GENERIC: Generic signature: <T, U where T : P1, U : P1, T.Assoc == U.Assoc>
```

Comment:
Yes, you are correct. Because I moved typo correction into the type checker, we're no longer able to feed that result (the correction of `T.assoc` to `T.Assoc`) back into the `GenericSignatureBuilder` to allow for correcting member types of `T.Assoc` nor can we have that affect the resulting generic signature. It's *possible* that we could do some kind of reprocessing or externally-introduced renaming to feed this back into the `GenericSignatureBuilder`, but I haven't tried that.